lib/TikZImage.pm

```diff
@@ -219,4 +247,47 @@ sub draw {
 	return $data;
 }
 
+sub warnLaTeXOutput {
+	my $self = shift;
+	my $working_dir = shift;
+	my $file = shift;
+	my $stdout = ($self->svgMethod eq 'dvisvgm') ? 'latex.stdout' : 'pdflatex.stdout';
+	warn "File $file was not created.";
+	if (open(my $err_fh, "<", "$working_dir/$stdout")) {
+		while (my $error = <$err_fh>) {
+			warn $error;
+		}
+		close($err_fh);
+	}
+}
+
+sub use_svgMethod {
+	my $self = shift;
+	my $working_dir = shift;
+	my $file = shift;
+	my $svgfile = $file =~ s/\.(dvi|pdf)$/\.svg/r;
+	if ($file =~ /\.dvi$/) {
+		system WeBWorK::PG::IO::externalCommand('dvisvgm') .
+			" $working_dir/$file --no-fonts --output=$working_dir/$svgfile > /dev/null 2>&1";
+	} else {
+		system WeBWorK::PG::IO::externalCommand($self->svgMethod) .
+			" $working_dir/$file $working_dir/$svgfile > /dev/null 2>&1";
+	}
+	warn "File $working_dir/$svgfile was not created" unless (-r "$working_dir/$svgfile");
+}
+
+sub use_convert {
+	my $self = shift;
+	my $working_dir = shift;
+	my $file = shift;
+	my $ext = shift;
+	my $outfile = $file =~ s/\.pdf$/\.$ext/r;
+	system WeBWorK::PG::IO::externalCommand('convert') .
+		join('',map {" -$_ " . $self->convertOptions->{input}->{$_}} (keys %{$self->convertOptions->{input}})) .
+		" $working_dir/$file" .
+		join('',map {" -$_ " . $self->convertOptions->{output}->{$_}} (keys %{$self->convertOptions->{output}})) .
+		" $working_dir/$outfile > /dev/null 2>&1";
+	warn "File $working_dir/$outfile was not created" unless (-r "$working_dir/$outfile");
```

Comment:
In the `use_svgMethod` and `use_convert` subroutines, there is no need for the `$file` parameters.  The value of `$file` is always going to be `image.$ext`.  Just use "image", which is again just an internal name.  I suggest changing to the following that does the job, conveys the pertinent information if errors occur, and is cleaner.  Of course you also need to adjust the calls to these above.
```suggestion
sub use_svgMethod {
	my $self = shift;
	my $working_dir = shift;
	if ($self->svgMethod eq 'dvisvgm') {
		system WeBWorK::PG::IO::externalCommand('dvisvgm') .
			" $working_dir/imag.dvi --no-fonts --output=$working_dir/image.svg > /dev/null 2>&1";
	} else {
		system WeBWorK::PG::IO::externalCommand($self->svgMethod) .
			" $working_dir/image.pdf $working_dir/image.svg > /dev/null 2>&1";
	}
	warn "Failed to generate svg file." unless -r "$working_dir/image.svg";
}

sub use_convert {
	my $self = shift;
	my $working_dir = shift;
	my $ext = shift;
	system WeBWorK::PG::IO::externalCommand('convert') .
		join('',map {" -$_ " . $self->convertOptions->{input}->{$_}} (keys %{$self->convertOptions->{input}})) .
		" $working_dir/image.pdf" .
		join('',map {" -$_ " . $self->convertOptions->{output}->{$_}} (keys %{$self->convertOptions->{output}})) .
		" $working_dir/image.$ext > /dev/null 2>&1";
	warn "Failed to generate $ext file." unless -r "$working_dir/image.$ext";
}
```

Reply:
I would commit this directly but it should be paired with the changes in the calls above. So I committed to my local branch.

Reply:
Yeah, I should have also made the suggestions in the calls above.  Then you could have added them as a batch and committed them.  I was just playing with this suggestion feature that @drdrew42 pointed out.